### ISSUE TITLE
`MathTex` not working

### ISSUE THREAD

**Issue by Wizorx** [maloyan/manim-js#396]
## Description
When trying to add a `MathTex`object to a scene, it doesn't appear

## Steps to Reproduce
1. Create a `MathTex` object
2. Add it to the scene

## Expected Behavior
For the object to be visible in the scene

## Actual Behavior
It's not visible

## Environment
- Browser: Brave 1.90.122
- OS: Windows 11 10.0.26200
- manim-web version: 0.3.21

## Additional Context
Here's the console output and the code that generated it:

```
three.core.js:2001 THREE.Material: parameter 'emissive' has value of undefined.
warn @ three.core.js:2001
ParseMethods-D3_HeS1K.js:1415 MathJax Warning: Package 'configmacros' not found.  Omitted.
warn @ ParseMethods-D3_HeS1K.js:1415
ParseMethods-D3_HeS1K.js:1415 MathJax Warning: Package 'newcommand' not found.  Omitted.
warn @ ParseMethods-D3_HeS1K.js:1415
ParseMethods-D3_HeS1K.js:1415 MathJax Warning: Package 'ams' not found.  Omitted.
warn @ ParseMethods-D3_HeS1K.js:1415
src-CW4pfZdP.js:49506 MathTex rendering error: Error: Can't find handler for document
    at f.handlesDocument (mathjax-7LgaTjtj.js:49:9)
    at f.document (mathjax-7LgaTjtj.js:52:15)
    at Object.document (mathjax-7LgaTjtj.js:113:21)
    at eo (src-CW4pfZdP.js:25261:47)
    at async yl._render (src-CW4pfZdP.js:49510:11)
    at async yl.waitForRender (src-CW4pfZdP.js:49481:30)
(anonymous) @ src-CW4pfZdP.js:49506
src-CW4pfZdP.js:38803 Scene: async mobject render failed for yl Error: Can't find handler for document
    at f.handlesDocument (mathjax-7LgaTjtj.js:49:9)
    at f.document (mathjax-7LgaTjtj.js:52:15)
    at Object.document (mathjax-7LgaTjtj.js:113:21)
    at eo (src-CW4pfZdP.js:25261:47)
    at async yl._render (src-CW4pfZdP.js:49510:11)
    at async yl.waitForRender (src-CW4pfZdP.js:49481:30)
(anonymous) @ src-CW4pfZdP.js:38803
spherical coordinates.html:1 Unsafe attempt to load URL file:///D:/OneDrive/Desktop/spherical%20coordinates.html from frame with URL file:///D:/OneDrive/Desktop/spherical%20coordinates.html. 'file:' URLs are treated as unique security origins.
```

```
<!DOCTYPE html>
<html>
<head>
  <style>
    * { margin: 0; padding: 0; box-sizing: border-box; }
    body { background: #191919; display: flex; justify-content: center; align-items: center; width: 100vw; height: 100vh; overflow: hidden; }
    #container { width: 100vw; height: 100vh; }
  </style>
</head>
<body>
  <div id="container"></div>

  <script type="module">
    import { Dot3D, YELLOW, makeDraggable, MathTex, Text, ThreeDAxes, Axes, ThreeDScene, UP, DOWN, OUT, Line3D, Angle, WHITE, DashedLine, PolarPlane, Sphere} from 'https://esm.sh/manim-web@0.3.21';


    const scene = new ThreeDScene(document.getElementById('container'), {
      width: window.innerWidth,
      height: window.innerHeight,
      backgroundColor: '#191919',
      phi: 75 * (Math.PI / 180),
      theta: -45 * (Math.PI / 180),
      distance: 20,
      fov: 30,
      enableOrbitControls: true,
      orbitControlsUp: 'z'
    });

    const axes = new ThreeDAxes({
      xRange: [-5, 5, 1],
      yRange: [-5, 5, 1],
      zRange: [-5, 5, 1],
      xLength: 10,
      yLength: 10,
      zLength: 10,
      showTicks: false,
    });

  
    const dot = new Dot3D({ radius: 0.1, color: YELLOW }).moveTo(axes.coordsToPoint(1,1,1)).addUpdater(() => {});
    makeDraggable(dot, scene,);

    const radius = new Line3D({start: [0,0,0], end: dot.getCenter(), lineWidth: 40}).addUpdater((mob) => {
      mob.setEnd(dot.getCenter());
    });


    const r = new MathTex({ latex: 'r', color: WHITE, fontSize: 22}).addUpdater((mob) => {
      mob.moveTo(radius.getMidpoint());
    });

    scene.add(axes, dot, radius,r );
    await scene.wait(Infinity);
  </script>
</body>
</html>
```

**Comment by maloyan** [comment]
## Root cause

The error `Can't find handler for document` is thrown by `mathjax.handlers.handlesDocument(...)` when the `handlers` list is empty for the `mathjax` singleton on which `mathjax.document(...)` is called.

`MathJaxRenderer.ts` issues 5 separate dynamic imports for MathJax submodules:

```ts
await import('@mathjax/src/js/mathjax.js');
await import('@mathjax/src/js/input/tex.js');
await import('@mathjax/src/js/output/svg.js');
await import('@mathjax/src/js/adaptors/liteAdaptor.js');
await import('@mathjax/src/js/handlers/html.js');
// + AmsConfiguration, NewcommandConfiguration, ConfigMacrosConfiguration
```

In the locally-built `dist/`, Vite/Rollup shares the `mathjax-*.js` chunk between callers (`dist/html-*.js` does `import { t as s } from "./mathjax-*.js"`), so `RegisterHTMLHandler` writes to the same singleton that `MathJax.document(...)` later reads.

**On esm.sh, that sharing breaks.** Inspecting `https://esm.sh/manim-web@0.3.21/es2022/dist/html-DU38zqaN.mjs` shows it has its **own** inlined `var yt = { handlers: new he(), ... }`. Meanwhile `mathjax-7LgaTjtj.mjs` has its own `var d = { handlers: new f(), ... }`. esm.sh re-bundles each `.mjs` independently and loses the shared-chunk identity, so two parallel `mathjax` singletons exist. `RegisterHTMLHandler` writes to `yt.handlers`; the caller queries `d.handlers`; empty → throw.

This is why MathTex works in local dev / direct `dist/` consumption but fails through `https://esm.sh/manim-web@...`.

## Reproduction

Confirmed by:
1. Inspecting esm.sh output (`/dist/html-*.mjs` has no import from the mathjax chunk, contains an inline `yt` singleton).
2. The error path `f.handlesDocument` in the user’s console trace maps to `HandlerList.handlesDocument`, which throws exactly when `items` is empty.

## Fix plan

Collapse all `@mathjax/src/...` access into a **single** internal module so esm.sh sees one chunk and cannot duplicate the singleton.

1. New file `src/mobjects/text/MathJaxBundle.ts` with *static* imports of every `@mathjax/src` entry currently used:
   ```ts
   import { mathjax } from '@mathjax/src/js/mathjax.js';
   import { TeX } from '@mathjax/src/js/input/tex.js';
   import { SVG } from '@mathjax/src/js/output/svg.js';
   import { liteAdaptor } from '@mathjax/src/js/adaptors/liteAdaptor.js';
   import { RegisterHTMLHandler } from '@mathjax/src/js/handlers/html.js';
   import '@mathjax/src/js/input/tex/ams/AmsConfiguration.js';
   import '@mathjax/src/js/input/tex/newcommand/NewcommandConfiguration.js';
   import '@mathjax/src/js/input/tex/configmacros/ConfigMacrosConfiguration.js';
   export { mathjax, TeX, SVG, liteAdaptor, RegisterHTMLHandler };
   ```
2. `MathJaxRenderer.ts`: replace the 5 dynamic imports with a single `await import('./MathJaxBundle.js')` so all `@mathjax/src` code lands in one chunk. The mathjax module is referenced once → singleton remains unique.
3. Stop externalizing `@mathjax/src` in `vite.config.ts` (or keep it externalized but verify the bundle keeps it as one chunk; first option is simpler).
4. **Self-healing fallback**: after `RegisterHTMLHandler(adaptor)`, probe `mathjax.handlers` length. If 0, throw a typed error that triggers strategy 2 (CDN script tag) — so future consumer bundlers that re-split still produce a working render rather than a silent failure.
5. Regression test (`src/mobjects/text/mathjax-render-regression.test.ts`): assert `mathjax.handlers.items.length > 0` after `renderLatexToSVG('x')`.
6. Integration: rebuild `dist/`, serve via local http with esm.sh-style chunk loading, run user’s exact HTML, confirm no error and `r` label renders.

Will branch off `main`, implement, run vitest + manual browser repro, and open a PR.
